File: go/for-loop.go
package main

import (
	"fmt"
	"os"
	"math/rand"
	"time"
)

func main() {

	generator := rand.New(rand.NewSource(time.Now().UnixNano()))
	solution := generator.Intn(100)+1

	fmt.Println("You have 10 tries to guess my number between 1 and 100.\n")

	for i := 1; i <= 10; i++ {
		fmt.Println(fmt.Sprintf("[%d/10] What's your guess?", i))
		var guess int
		fmt.Scanln(&guess)
		
		if guess < solution {
			fmt.Println("My number is bigger.")
		} else if guess > solution {
			fmt.Println("My number is smaller.")
		} else {
			fmt.Println("That's it. You guessed it.");
			os.Exit(0)
		}
	}

	fmt.Println("Ohh you didn't guess it. :(");
	os.Exit(1)
}

File: python/for-loop.py
import sys
import random

solution = random.randint(1, 100)

print("You have 10 tries to guess my number between 1 and 100.")

for i in range(10):
	print("[%d/10] What's your guess?" % (i+1))
	guess = int(sys.stdin.readline())

	if guess < solution:
		print("My number is bigger.");
	elif guess > solution:
		print("My number is smaller.");
	else:
		print("That's it. You guessed it.");
		exit(0)

print("Ohh you didn't guess it. :(")
exit(1)
